src/components/molecules/TablesControlBar/TablesControlBar.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useMemo, useCallback } from "react";
+import classNames from "classnames";
+
+import { Table } from "types/Table";
+import { User } from "types/User";
+import { AnyVenue } from "types/venues";
+
+import { experienceSelector } from "utils/selectors";
+import { WithId } from "utils/id";
+
+import { useSelector } from "hooks/useSelector";
+
+import { Checkbox } from "components/atoms/Checkbox";
+
+import "./TablesControlBar.scss";
+
+export interface TablesControlBarProps {
+  defaultTables: Table[];
+  isChecked: boolean;
+  venue: AnyVenue;
+  users: readonly WithId<User>[];
+  onToggleTables: (value: React.SetStateAction<Table[]>) => void;
+  className?: string;
+}
+
+export const TablesControlBar: React.FC<TablesControlBarProps> = ({
+  defaultTables,
+  isChecked,
+  venue,
+  users,
+  onToggleTables,
+  className,
+}) => {
+  const experience = useSelector(experienceSelector);
+
+  // not locked and not full tables
+  const freeTables = useMemo(
+    () =>
+      defaultTables
+        .filter((table) => !experience?.tables[table.title]?.locked)
+        .filter(
+          (table) =>
+            !users.filter(
+              (u) => u.data?.[venue.name]?.table === table.reference
+            ).length
+        ),
+    [defaultTables, experience?.tables, users, venue.name]
+  );
```

Comment:
moreover I believe this logic should be moved out of the `TableControlBar`.
Here we just define if the checkbox is checked or not, and then trigger the updates in the parent component and let it decided what to do with that information.

if you want not to repeat the logic. You can move that filtering function to some utils. 

<hr />

no need to run `filter` twice. You can do all the logic in the first one.

Reply:
Thank you for idea!
fixed in 5aa6329bd90551d39634b475d84adbb35e5e663a